bot.py

```diff
@@ -89,32 +97,78 @@ async def on_member_join(member):
     name="help",
     help="Displays the syntax and the description of all the commands.",
 )
-async def help_command(ctx):
+async def help_command(ctx, name=None):
     """Help command outputs description about all the commands"""
     try:
+        # typing indicator
+        async with ctx.typing():
+            await asyncio.sleep(1)
+
+        list_commands = []  # make dictionary to store all commands
+        for command in bot.commands:
+            # check if the user input specific command after help
+            if command.name == name:
+                # for commands that receive multiple parameter
+                if command.name in ["add", "remove"]:
+                    return await command(ctx)
+                return await command(ctx, "help")
+            list_commands.append(
+                {"command": f"{command.name}", "description": f"{command.help}"}
+            )
+        # sort commands alphabetically
+        list_commands = sorted(list_commands, key=lambda x: x["command"])
+
         embed_data = EmbedData()
         embed_data.title = f"How to use {bot.user.name}"
-        embed_data.description = (
-            f"`All Data from NA server`\n\n <@!{bot.user.id}> <command>"
-        )
+        embed_data.description = f"** **\n<@!{bot.user.id}> <command> --help\
+              \n\n`--help` shows the information of the command:"
         embed_data.color = discord.Color.gold()
 
         # ADD thumbnail (Image can be changed whatever we want. eg.our logo)
         embed_data.thumbnail = "https://emoji.gg/assets/emoji/3907_lol.png"
 
         embed_data.fields = []
-        embed_data.fields.append({"name": "** **", "value": "** **", "inline": False})
+        embed_data.fields.append(
+            {"name": "** **", "value": "**\nCommands:**", "inline": False}
+        )
 
-        for command in bot.commands:
-            if not str(command).startswith("help"):
+        for command in list_commands:
+            if command["command"] == "help":
+                embed_data.fields.append(
+                    {
+                        "name": "** **",
+                        "value": "**{}** `command_name`\n{}".format(
+                            command["command"],
+                            "Display the information of the command",
+                        ),
+                        "inline": False,
+                    }
+                )
+            elif command["command"] in ["list", "teams", "clear"]:
                 embed_data.fields.append(
                     {
                         "name": "** **",
-                        "value": f"<@!{bot.user.id}> **{command.name} summoner_name** \n \
-                            {command.help}",
+                        "value": "**{}**\n{}".format(
+                            command["command"], command["description"]
+                        ),
                         "inline": False,
                     }
                 )
+
+            else:
```

Comment:
what does this handle?

Reply:
This deals with the situation in which a `summoner_name` must be entered after the command.
Because each command has a different user input such as `rank` accepts `summoner_name` or `list` doesn't accept anything.